module/cloud-services/files/helper.js

```diff
@@ -26,86 +26,150 @@ module.exports = class FilesHelper {
+  static preSignedUrls(payloadData, referenceType, userId) {
     return new Promise(async (resolve, reject) => {
       try {
-          
-          let payloadIds = Object.keys(payloadData);
 
-          let bucketName = "";
-          let cloudStorage = process.env.CLOUD_STORAGE;
+        let payloadIds = Object.keys(payloadData);
 
-          if( cloudStorage === "AWS" ) {
-              bucketName = process.env.AWS_BUCKET_NAME;
-          } else if (cloudStorage === "GC" ) {
-            bucketName = process.env.GCP_BUCKET_NAME;
-          } else {
-            bucketName = process.env.AZURE_STORAGE_CONTAINER;
-          }
+        let bucketName = "";
+        let cloudStorage = process.env.CLOUD_STORAGE;
 
-          let result = {
-              [payloadIds[0]] : {}
-          };
-
-          if( referenceType === constants.common.PROJECT ) {
-              
-                for( let pointerToPayload = 0; pointerToPayload < payloadIds.length; pointerToPayload++ ) {
-                    
-                    let payloadId = payloadIds[pointerToPayload];
-                    let folderPath = "project/" + payloadId + "/" + userId + "/" + gen.utils.generateUniqueId();
-                    let imagePayload = 
-                    await filesHelpers.preSignedUrls(
-                        payloadData[payloadId].files,
-                        bucketName,
-                        cloudStorage,
-                        folderPath
-                    );
-
-                    if( !imagePayload.success ) {
-                        return resolve({
-                            status : httpStatusCode['bad_request'].status,
-                            message : constants.common.FAILED_PRE_SIGNED_URL,
-                            result : {}
-                        });
-                    }
-
-                    if( !result[payloadId] ) {
-                        result[payloadId] = {};
-                    }
-
-                    result[payloadId]["files"] = imagePayload.result;
-                }
+        if (cloudStorage === "AWS") {
+          bucketName = process.env.AWS_BUCKET_NAME;
+        } else if (cloudStorage === "GC") {
+          bucketName = process.env.GCP_BUCKET_NAME;
+        } else {
+          bucketName = process.env.AZURE_STORAGE_CONTAINER;
+        }
 
-          } else {
-            let folderPath = "survey/" + payloadIds[0] + "/" + userId + "/" + gen.utils.generateUniqueId();
-            
-            let imagePayload = await filesHelpers.preSignedUrls(
-                payloadData[payloadIds[0]].files,
+        let result = {
+          [payloadIds[0]]: {}
+        };
+
+        if (referenceType === constants.common.PROJECT) {
+
+          for (let pointerToPayload = 0; pointerToPayload < payloadIds.length; pointerToPayload++) {
+
+            let payloadId = payloadIds[pointerToPayload];
+            let folderPath = "project/" + payloadId + "/" + userId + "/" + gen.utils.generateUniqueId();
```

Comment:
@aman-tunerlabs let's change the structure to {{survey/reports/observation/projects}}/{{projectId/surveySubmissionId/observationSubmissionId}}/uuid.{{fileExtension}}